Source/WebCore/html/parser/HTMLConstructionSite.cpp

```diff
@@ -262,7 +262,7 @@ HTMLConstructionSite::HTMLConstructionSite(Document& document, OptionSet<ParserC
     , m_maximumDOMTreeDepth(maximumDOMTreeDepth)
     , m_inQuirksMode(document.inQuirksMode())
 {
-    ASSERT(m_document.isHTMLDocument() || m_document.isXHTMLDocument());
+    ASSERT(document.isHTMLDocument() || document.isXHTMLDocument());
```

Comment:
Why not `document()`, like in the version taking the `DocumentFragment` argument?

Reply:
Unlike in the other constructor, we have a `document` argument here that we can leverage. Note that `document()` wouldn't build since there is a `document` parameter. You'd have to use `this->document()`. I think `document` looks much better than `this->document()`.